packages/integration-tests/src/test-in-memory/integration_tests.test.ts

```diff
@@ -1,10 +1,75 @@
-import { runCDKSnapshotTestSuite } from '../cdk_snapshot_test_runner.js';
-import { fromConventionalDir } from '../cdk_snapshot_test_suite_generator.js';
-import * as path from 'path';
-
-runCDKSnapshotTestSuite(
-  'CDK snapshot tests',
-  fromConventionalDir(path.join('..', 'test-projects')).concat(
-    fromConventionalDir(path.join('..', '..', 'create-amplify', 'templates'))
-  )
-);
+import { dataStorageAuthWithTriggers } from '../test-projects/data-storage-auth-with-triggers-ts/amplify/test_factories.js';
+import {
+  assertStableLogicalIds,
+  defineBackendTemplateHarness,
+} from '../define_backend_template_harness.js';
+import { it } from 'node:test';
+import { dataWithoutAuth } from '../test-projects/standalone-data-auth-modes/amplify/test_factories.js';
+import { dataWithoutAuthNoAuthMode } from '../test-projects/standalone-data-sandbox-mode/amplify/test_factories.js';
+
+/**
+ * This test suite is meant to provide a fast feedback loop to sanity check that different feature verticals are working properly together.
+ * Specific feature configurations should be checked at the unit test level.
+ * Some assertions about how feature verticals interact could be appropriate here.
+ * Critical path interactions should be exercised in a full e2e test.
+ */
+
+void it('data storage auth with triggers', () => {
+  const templates = defineBackendTemplateHarness(dataStorageAuthWithTriggers);
+
+  assertStableLogicalIds(templates.root, 'AWS::CloudFormation::Stack', [
+    'auth179371D7',
+    'data7552DF31',
+    'function1351588B',
+    'storage0EC3F24A',
+  ]);
+
+  assertStableLogicalIds(templates.auth, 'AWS::Cognito::UserPool', [
+    'amplifyAuthUserPool4BA7F805',
+  ]);
+
+  assertStableLogicalIds(templates.data, 'AWS::AppSync::GraphQLApi', [
+    'amplifyDataGraphQLAPI42A6FA33',
+  ]);
+  assertStableLogicalIds(templates.data, 'AWS::CloudFormation::Stack', [
+    'amplifyDataAmplifyTableManagerNestedStackAmplifyTableManagerNestedStackResource86290833',
+    'amplifyDataFunctionDirectiveStackNestedStackFunctionDirectiveStackNestedStackResource1246A302',
+    'amplifyDataTodoNestedStackTodoNestedStackResource551CEA56',
+  ]);
+
+  assertStableLogicalIds(templates.storage, 'AWS::S3::Bucket', [
+    // eslint-disable-next-line spellcheck/spell-checker
+    'amplifyStorageamplifyStorageBucketC2F702CD',
+  ]);
```

Comment:
One benefit of doing this that I can see is that if we load test project from directory and expect assertions to be defined then this would force new test project authors to add some assertions.

Reply:
We aren't loading these tests from a directory though, it's just a normal `import`. I think at this point any class would look essentially the same as the tests, just with more steps.

Reply:
My comment was assuming we go back to scanning and loading test projects from directory.

But the last sentence made me thing that perhaps we could split these tests and embed them in test project directory. I.e. decentralize this.

Feel free to proceed without solving this discussion (as this is potential YAGNI as you said). If you think it's worth recording the discussion please cut GH issue.

Reply:
I'm going to leave this for now. I think it will organically come up if/when the integration tests start to have repeated bits that can be pulled into central components.